sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -422,6 +426,15 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
             }
             metrics.foreach(m => checkMetric(m, m))
 
+          // see Analyzer.ResolveMelt
+          case m: Melt if m.childrenResolved && m.ids.forall(_.resolved) && m.values.isEmpty =>
+            failAnalysis("MELT_REQUIRES_VALUE_COLUMNS", Array(m.ids.mkString(", ")))
```

Comment:
I think toSQLId() is need to make sure the ids are properly decorated with back-ticks.